frontend/javascripts/oxalis/model/reducers/skeletontracing_reducer.js

```diff
@@ -252,6 +252,28 @@ function SkeletonTracingReducer(state: OxalisState, action: Action): OxalisState
             .getOrElse(state);
         }
 
+        case "SET_NODE_POSITION": {
+          const { position, nodeId, treeId } = action;
+          return getNodeAndTree(skeletonTracing, nodeId, treeId)
+            .map(([tree, node]) => {
+              const diffableMap = skeletonTracing.trees[tree.treeId].nodes;
+              const newDiffableMap = diffableMap.set(
+                node.id,
+                update(node, { position: { $set: position } }),
+              );
+              return update(state, {
```

Comment:
Could you use `updateKey` and `updateKey4` (?) from `deep_update` instead of the update function? See this issue for context: https://github.com/scalableminds/webknossos/issues/3055

I know that the issue is quite old (and hidden), but I'd find it cool when new code uses the type-checked version for updating things :)

Reply:
Unfortunate, using `updateKey4` here isn't possible because the flow definition expects that all key must exist and cannot be null or undefined, but the `skeleton` property is optional and thus can be undefined/null.

FYI: no `updateKeyX` function is used in the whole `skeletontracing_reducer`, thus making this change quite some work.

Is it fine with you to ignore this suggested change under these circumstances @philippotto?

Reply:
Oh, ok, then nevermind :)